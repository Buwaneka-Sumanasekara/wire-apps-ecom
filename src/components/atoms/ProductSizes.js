import React, { useCallback, useEffect, useState } from 'react';
import { Text, TouchableOpacity, View } from 'react-native';




const ProductSizes = ({ sizes,selectedSize:passedSelectedSize,onChange }) => {
    const [selectedSize, setSelectedSize] = useState(passedSelectedSize || null);



    useEffect(() => {
        onChange(selectedSize);
    }, [selectedSize,onChange])


    const handleSizeSelection = useCallback((size) => {
        setSelectedSize(size);
    }, [sizes])

    return (
        <View className={"mt-3"}>
            <Text>{"Choose size"}</Text>
            <View className={"flex flex-row flex-wrap gap-2 w-screen mt-1"}>
                {(sizes || []).map((size) => {
                    const isSelected = (selectedSize === size);
                    return (
                        <TouchableOpacity
                            key={`size-id-${size}`}
                            onPress={() => handleSizeSelection(size)}
                            className={`${(isSelected ? `bg-black` : `bg-slate-100`)} p-3 rounded-lg`}>
                            <Text className={(isSelected ? "text-white" : "text-black")}>{size}</Text>
                        </TouchableOpacity>
                    )
                })}
            </View>
        </View>
    )

}

export default ProductSizes;