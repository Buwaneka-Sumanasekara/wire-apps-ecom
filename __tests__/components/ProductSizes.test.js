import React from 'react';
import renderer from 'react-test-renderer';
import ProductSizes from '../../src/components/atoms/ProductSizes';


describe('<ProductSizes />', () => {
    it('rendered sucessfully', () => {
        const tree = renderer.create(<ProductSizes />).toJSON()
        expect(tree).toMatchSnapshot()
    });

});

