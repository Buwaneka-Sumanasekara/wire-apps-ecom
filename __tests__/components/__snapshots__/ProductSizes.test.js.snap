// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductSizes /> rendered sucessfully 1`] = `
<View
  style={
    [
      {
        "marginTop": 12,
      },
    ]
  }
>
  <Text>
    Choose size
  </Text>
  <View
    style={
      [
        {
          "display": "flex",
        },
        {
          "flexDirection": "row",
        },
        {
          "flexWrap": "wrap",
        },
        {
          "marginLeft": -8,
          "marginTop": -8,
        },
        {
          "width": 750,
        },
        {
          "marginTop": 4,
        },
      ]
    }
  />
</View>
`;
